### PR TITLE
Fixes #309

### LINKED ISSUE #309: FakeSign and IBCMerge together break incremental build
Repro steps:
1. cd Open\src\Compilers\Core\Portable
2. MSBuild /m /p:Configuration=Release
3. Repeat step 2.
Result:
Build fails with: 
  EXEC : error : The module D:\Roslyn\Open\src\Compilers\Core\Portable\obj\Release\Microsoft.CodeAnalysis.dll, already contains an IBC resource. [D:\Roslyn\Open\src\Compilers\Core\Portable\CodeAnalysis.csproj]

The problem is that the FakeSign step has modified the DLL after IBCMerge, thwarting the .timestamp file that each step writes out.

We need to come up with a technique so that multiple 'rewriters' can modify the binary and keep incremental builds working.

### PR DESCRIPTION
**Pull request by pharring**
We have two targets that each modify a compiled binary in-place:
ApplyOptimizations and FakeSign. They each write out their own sentinel
file to help keep incremental build healthy and quick. However, this
causes problems when you need to run both (e.g. on a developer box
compiling Release binaries).
This change moves the sentinel file into a third task that depends on
the other two. I hope it's obvious how to extend this if more "post
compile binary modification" targets are added later.

### CODE REVIEW

**Comment by pharring** [comment]
@jasonmalinowski Please take a look.

File: build/VSL.Imports.Closed.targets [hunk 1]
```diff
@@ -1,4 +1,35 @@
...
+          DependsOnTargets="ApplyOptimizations;FakeSign"
+          Inputs="@(IntermediateAssembly)"
+          Outputs="@(IntermediateAssembly);$(PostCompileBinaryModificationSentinelFile)">
+
+    <!-- Write out a sentinel timestamp file to prevent unnecessary work in incremental builds. -->
+    <Touch AlwaysCreate="true" Files="$(PostCompileBinaryModificationSentinelFile)" />
+  </Target>
+
+  <Target Name="CleanPostCompileBinaryModification">
+    <Delete Files="$(PostCompileBinaryModificationSentinelFile)" />
```
Comment (jasonmalinowski) [inline build/VSL.Imports.Closed.targets]: The easier option is to put the ...SentinelFile item into the "FileWrites" item group. Those get stored and clean will delete it automatically.

File: build/VSL.Imports.Closed.targets [hunk 2]
```diff
@@ -13,25 +44,44 @@
...
+  </PropertyGroup>
 
-  <Target Name="CleanApplyOptimizations">
-    <Delete Files="@(IntermediateAssembly->'%(RootDir)%(Directory)%(Filename).ibcmerge.timestamp')" />
+  <Target Name="FakeSign"
+          Condition="'$(DelaySign)' == 'true' AND '$(ShouldSignBuild)' != 'true' AND ('$(Language)' == 'C#' OR '$(Language)' == 'VB')"
+          Inputs="@(IntermediateAssembly)"
+          Outputs="@(IntermediateAssembly);$(PostCompileBinaryModificationSentinelFile)">
+    
+    <Exec Command="$(VSLOutDir)\FakeSign.exe &quot;$(IntermediateOutputPath)$(TargetFileName)&quot;" />
```
Comment (jasonmalinowski) [inline build/VSL.Imports.Closed.targets]: Re-use @(IntermediateAssembly) just so it's clear the Exec is on the same file as the Target Inputs/Outputs?

**Comment by jasonmalinowski** [comment]
:+1:

**Comment by pharring** [comment]
@jasonmalinowski Thanks for the review. I applied both your suggested changes.

**Pull request merged**